src/core/control/Control.cpp

```diff
@@ -1779,7 +1779,28 @@ void Control::undoRedoPageChanged(PageRef page) {
     }
 }
 
-void Control::selectTool(ToolType type) {
+void Control::handleSelectToolAction(ToolType type) {
+    // keep text-selection when switching from text to seletion tool
+    auto oldTool = getToolHandler()->getActiveTool();
+    if (oldTool && win
+                && (type == TOOL_SELECT_RECT || type == TOOL_SELECT_REGION || type == TOOL_SELECT_OBJECT)
```

Comment:
nit: instead of directly checking `type == TOOL_SELECT_RECT || ...`, it would be better to define this expression as a function `isSelectToolType(ToolType)` in `src/core/control/ToolEnums.h`, and then call the function here. This would allow us to potentially add additional selection tools in the future without needing to change this part of the code.